src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1207 @@
+
+    def __eq__(self, other):
+        r"""
+        Check if ``self`` and ``other`` are equal. This means checking that they
+        are equivalent.
+
+        TESTS::
+
+            sage: var('x1')
+            x1
+            sage: S = LPASeed({x1:2},base_ring=ZZ)
+            sage: T = S.mutate(0, inplace=False)
+            sage: T==S
+            False
+        """
+
+        # we check if cluster variables are is_equivalent of each other,
+        # up to unit multiple. Since clusters determine the seed,
+        # this is sufficient
+        return self.is_equivalent(other)
+
+    def __copy__(self):
+        r"""
+        Returns a copy of ``self``.
+
+        TESTS::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: T1 = S._copy_()
+            sage: T1.mutate(0, inplace=True)
+            sage: T2 = S.mutate(0, inplace=False)
+            sage: T1==T2
+            True
+        """
+        return LPASeed(self)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: print(LPASeed({x1: 1 + x2, x2: 1 + x1}))
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+        """
+        return ("A seed with cluster variables {0}"
+                " and exchange polynomials {1}").format(self._cluster_vars,
+                                                        self._exchange_polys)
+
+    def __hash__(self):
+        """
+        Return a hash of ``self``.
+
+        EXAMPLES:
+
+        We check that two seeds with the same hash are equal::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: T = LPASeed(S)
+            sage: S.mutate([0,1,0])
+            sage: T.mutate([1,0])
+            sage: hash(S) == hash(T)
+            True
+            sage: S == T
+            True
+        """
+
+        return hash(frozenset(self.cluster()))
+
+    def _latex_(self):
+        r"""
+        Returns a `\LaTeX` representation of this seed.
+
+        TESTS::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: latex(LPASeed({x1: 1 + x2, x2: 1 + x1}))
+            \left(\left(x_{1}, x_{2} + 1\right), \left(x_{2}, x_{1} + 1\right)\right)
+
+        """
+        cvars = self.cluster()
+        polys = self.exchange_polys()
+        cvar_poly_pairs = list()
+        for i in range(self.rank()):
+            cvar_poly_pairs.append(latex(tuple([cvars[i], polys[i]])))
+
+        return latex(tuple(cvar_poly_pairs))
+
+    # private method to simplify mutation sequence given list L of indices
+    # we essentially apply the involution rule until no consecutive entries
+    # are the same
+
+    @staticmethod
+    def _remove_repeat_indices(L):
```

Comment:
Why is this a `@staticmethod`? It seems like it should be a normal function.

Reply:
> Why is this a `@staticmethod`? It seems like it should be a normal function.

This is because it is only used as a helper function for formatting the sequence of mutations. I want nothing else to use it or depend on its implementation.

Reply:
IMO, doing it this way is not a good programming practice. The `@staticmethod` would be good if it was something that depended on the class, but this is really just a standalone helper function. So it really should be pulled out as a standalone function (although still with the same name so it is hidden from the doc).

Reply:
I don't really want this method to pollute Sage's global namespace when I only intend for it to be used for formatting the sequence of mutations, but I will trust your judgement.

Reply:
Just having a separate function within a file will not pollute the global namespace. You're doing the correct thing and only specifically importing to the `all.py` file what should be in the global namespace. You will see for the doctests that you will need to explicitly import the function.